### PR TITLE
Communities join screens - add toast after joining/leaving

### LINKED ISSUE #14458: Communities join screens (Non token gated) - Add toast after joining/leaving
Depends on: https://github.com/status-im/status-mobile/issues/13831

Integrate the toast component from quo2 into the designs for community join screens.
This should only show directly after a user has joined a community but not be shown again once the user navigates away and back to the community overview page.

See the designs here:
https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?node-id=1386%3A110589&t=4pBzp6AcBrgJRPmM-0

### PR DESCRIPTION
**Pull request by alwx**
fixes #14458

### Summary

Adds toasts for cases when joining and leaving the community (+ when a request to join the community is sent).
Doesn't include translations for texts.

#### Platforms

- Android
- iOS

status: ready

### CODE REVIEW

File: src/status_im/communities/core.cljs
Comment (J-Son89) [inline src/status_im/communities/core.cljs]: should this color come from quo2/foundations ?

**Review by J-Son89 — APPROVED** [review]

File: src/status_im/communities/core.cljs
Comment (flexsurfer) [inline src/status_im/communities/core.cljs]: better to do not use `:dispatch-n`, we should keep `handle-response` and also use handler for `toasts/create` as well

File: src/status_im/communities/core.cljs
Comment (flexsurfer) [reply src/status_im/communities/core.cljs]: yes color should be a property in the component not the color value

**Comment by alwx** [comment]
Some update:
`:toasts/create` event was using `dispatch` inside without any reason for it, and was doing basically the same as `:toasts/upsert` except it was assigning `id` automatically. 
I removed any need in separate `:toasts/create` and extended the functionality of `:toasts/update` to work without an explicitly specified id => made it possible to use the handler for `toasts/upsert` to display community join toast.

cc @flexsurfer

**Comment by yqrashawn** [comment]
> Some update: `:toasts/create` event was using `dispatch` inside without any reason for it, and was doing basically the same as `:toasts/upsert` except it was assigning `id` automatically. I removed any need in separate `:toasts/create` and extended the functionality of `:toasts/update` to work without an explicitly specified id => made it possible to use the handler for `toasts/upsert` to display community join toast.
> 
> cc @flexsurfer

Hi @alwx, the idea behind this is that most of the time the `:toasts/create` event will be used since there won't be additional logic behind most toasts and dev can just create it and call it a day.
`:toasts/upsert` is there to make the `id` explicit when we want more control on the toast.

Anyway, I'm fine with this change. There's a newly merged commit https://github.com/status-im/status-mobile/commit/02a1c3597f0f775bb1c9dd358ef655ea56b9b590 with usage of `:toasts/upsert`, can you update the usage there as well? You may need to update the test as well.

**Comment by VolodLytvynenko** [comment]
hi @alwx I noticed that some integration tests failed in this PR. Could you check and fix it? https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-14735/6/consoleText

Also, rebase this PR, please

**Comment by VolodLytvynenko** [comment]
@alwx Take a look also. e2e tests are failed. It is caused by the following issue: 

**ISSUE 1: The device lagged and does not react after the 'join' button is tapped on the community**
**Steps to reproduce:**
1. USER_A create community
2. Invite USER_B
3. Go to the 1-1 chat as USER_B
4. Tap 'view' button on the invitation
5. Tap 'join'

**Actual result:**

https://user-images.githubusercontent.com/52490791/211595411-b30e3744-5d58-4f7a-b4bf-66422fef0dee.mp4

The app is not reacting after 'join' button is tapped on the community

**Expected result:**
The user is joined the community and able to navigate further

**ENV:** any Android

**Logs:**
[Status.log](https://github.com/status-im/status-mobile/files/10384457/Status.log)
[geth.log](https://github.com/status-im/status-mobile/files/10384458/geth.log)

**Comment by VolodLytvynenko** [comment]
Hi, @J-Son89 Take a look at the issue https://github.com/status-im/status-mobile/issues/14659 this is not so serious  blocker for the current PR. But if you may fix it soon it'll be super! :)

**Comment by J-Son89** [comment]
> Hi, @J-Son89 Take a look at the issue #14659 this is not so serious blocker for the current PR. But if you may fix it soon it'll be super! :)

sure thing, I'll get on it this week!

**Comment by pavloburykh** [comment]
Hi @alwx ! Thanx for your work. I see that current PR has been merged into develop before it passed manual QA review. 

I guess you just didn't have a chance to look through current PR workflow process yet. That's ok) All detailed info about pipeline process can be found [here](https://notes.status.im/M73LlV1gT3OJ8kdSMY-ZQQ?view#Pipeline-process) 

In a few words, PR can be merged into develop when it has already passed manual QA and has label Tested - OK. QA team moves PR to the MERGE column if PR has been tested and no issues were found/ or all of found issues have been fixed. After developer merges PR and it is automatically been moved to DONE column. 

Currently we see that @VladimrLitvinenko has started testing and logged first [issue](https://github.com/status-im/status-mobile/pull/14735#issuecomment-1377480062) Other issues might be also found when testing will be continued. 

As PR has been already merged into develop - testing will be continued in nightly builds and PR related bugs will be logged into separate ticket. **cc** @VladimrLitvinenko

**Pull request merged**
